src/comm/TcpAcceptor.cc

```diff
@@ -399,12 +399,20 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     Ip::Address::FreeAddr(gai);
 
     // Perform NAT or TPROXY operations to retrieve the real client/dest IP addresses
-    if (conn->flags&(COMM_TRANSPARENT|COMM_INTERCEPTION) && !Ip::Interceptor.Lookup(details, conn)) {
-        debugs(50, DBG_IMPORTANT, "ERROR: NAT/TPROXY lookup failed to locate original IPs on " << details);
-        // Failed.
-        PROF_stop(comm_accept);
-        return Comm::COMM_ERROR;
-    }
+    /* NP: try TPROXY first, its much quieter than NAT when non-matching */
```

Comment:
I see no "first A, then B" tries here. I only see "either A or B". If the official code works like the current PR code -- either/or -- then we should remove the above comment. If the current PR code is correct, then we should also remove that comment. If the comment is correct, then the PR code needs to be changed accordingly.